File: maximum_subarray.cpp
// https://leetcode.com/problems/maximum-subarray/

class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int a = nums[0];
        int b = nums[0];
        for (int i = 1; i < nums.size(); i++) {
            a = std::max(nums[i], a + nums[i]);
            b = std::max(a, b);
        }
        return b;
    }
};

File: maximum_subarray.py
# https://leetcode.com/problems/maximum-subarray/

class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        best = nums[0]
        sum = nums[0]
        for i in range(1, len(nums)):
            sum = max(nums[i], sum + nums[i])
            best = max(best, sum)

        return best
